UmpleToJava/UmpleTLTemplates/JavaClassGenerator.ump

```diff
@@ -98,17 +98,27 @@ public <<# if (uClass.getIsAbstract()) { append(realSb, "{0} ", "abstract"); } #
 for (StateMachine smq : uClass.getStateMachines())
   {
     if (smq.isQueued())
-    {
-      append(realSb," implements Runnable");
+    { 
+      if (uClass.hasParentInterface() == false){
```

Comment:
@amidzak sorry about that. I completely forgot we talked about it before. 
The issue is not related to your PR. You are good to go.